src/stores/WalletStore.js

```diff
@@ -130,132 +126,59 @@ export default class WalletStore {
 
     // If setting Naka Wallet's account for the first time or the address changes, fetch the BOT balance right away.
     // After the initial BOT balance fetch, it will refetch on every new block.
-    const fetchInitBotBalance = isEmpty(this.addresses);
+    const fetchInitNbotBalance = isEmpty(this.addresses);
```

Comment:
you also need to fetch Naka balance too. `web3.eth.getBalance()`

Reply:
naka balance is updated as nakastore has a update listener  at line 22 in `NakaStore`